### PR TITLE
VMware SD-WAN and SASE Solution v1.0.0

### PR DESCRIPTION
**Pull request by stlaszlo**
Change(s):

Added new solution: VMware SD-WAN and SASE (v1.0.0)
Included Function App for VMware Edge Cloud Orchestrator API integration
Included Analytics Rules for VMware SD-WAN and SASE
Included Hunting Queries for VMware SD-WAN and SASE
Included Workbook for VMware SD-WAN and SASE
Reason for Change(s):

New Solution Added
Version Updated:

No
Testing Completed:

Yes
Checked that the validations are passing and have addressed any issues that are present:

Need Help

### CODE REVIEW

**Comment by v-prasadboke** [comment]
Hello @stlaszlo, Workbook images should be with dark and white background images. Please refer to these images Workbooks/Images/Preview for more clarification. 
Also add workbook metadata to this file Workbooks/WorkbooksMetadata.json

**Comment by v-prasadboke** [comment]
Hello @stlaszlo Hope you are doing well, Dropping this comment to check if you have any updates or need any help on the above requested changes. 
Thanks.

**Comment by stlaszlo** [comment]
@v-prasadboke thanks for checking in, I am actively working on this and will get an update tonight.

**Comment by v-prasadboke** [comment]
Thanks @stlaszlo, I'll wait for your updates.

**Comment by v-prasadboke** [comment]
Hello @stlaszlo, Thanks for committing the requested changes. I'll take a look at it and come back to you by 24 January, 2024

**Comment by v-prasadboke** [comment]
Hello @stlaszlo, Sorry for the inconvenience but due to technical reason we will provide you the update by 30 January, 2024.

**Comment by v-prasadboke** [comment]
Hello @stlaszlo, Can you provide me write access to your repo.

**Comment by stlaszlo** [comment]
@v-prasadboke you should be set as a collaborator, please go ahead and check.

File: Solutions/VMware SD-WAN and SASE/SolutionMetadata.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+	"publisherId": "velocloud",
+	"offerId": "azure-sentinel-solution-vmwaresase",
+	"firstPublishDate": "2023-12-31",
+	"providers": ["VMware"],
+	"categories": {
+		"domains" : ["Security - Threat Protection"],
+		"verticals": []
+	},
+	"support": {
```
Comment (v-prasadboke) [inline Solutions/VMware SD-WAN and SASE/SolutionMetadata.json:10]: Just wanted to confirm Microsoft team provide support to this solution

Comment (stlaszlo) [reply Solutions/VMware SD-WAN and SASE/SolutionMetadata.json:10]: It should be partner supported.

**Comment by v-prasadboke** [comment]
Please change the Support details also the zip is missing with python packages which are mentioned in requirements.txt file please add them. 
Please refer to this zip for more clarification 
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Cloudflare/Data%20Connectors/CloudflareConn.zip
Please go through the zip you will see python packages folder which includes python required libraries.

**Comment by v-prasadboke** [comment]
Hello @stlaszlo, please proceed with the above mentioned changes.

**Comment by stlaszlo** [comment]
Hello @v-prasadboke , ACK, work in progress, it will be pushed tomorrow, thanks.

**Comment by v-prasadboke** [comment]
> Hello @v-prasadboke , ACK, work in progress, it will be pushed tomorrow, thanks.

Thank you for the humble reply @stlaszlo.

**Comment by v-atulyadav** [comment]
Hi @stlaszlo,
To follow up appropriately, we would appreciate some indication of the timeframe for these changes. Thanks

**Comment by stlaszlo** [comment]
Hi @v-atulyadav and @v-prasadboke apologies, I was on sick leave for a few days, I will get the updates in today or early next week.

**Comment by v-prasadboke** [comment]
Hello @stlaszlo, Thanks for the responde. We will be waiting for the commits.

**Comment by stlaszlo** [comment]
Thanks @v-prasadboke requested changes committed.

**Comment by v-prasadboke** [comment]
Hello @stlaszlo, 
I have added python packages in the zip which were mentioned in requirements.txt 
Please add more if required.

**Comment by v-prasadboke** [comment]
Also can you provide me credentials to test function app
you can mail me them at v-prasadboke@microsoft.com

**Comment by stlaszlo** [comment]
@v-prasadboke sent credentials via email with some instructions, but should you need anything, please let me know, I am happy to push this through the finish line.

**Comment by v-prasadboke** [comment]
Hello @stlaszlo, can you share working images of function app

**Comment by v-prasadboke** [comment]
Hello @stlaszlo, 
There's an issue with analytic rule count. In analytic blade there shows no results for these 3 Analytic rules.
1. VMware SD-WAN Edge - Device Congestion Alert - Packet Drops
2. VMware SD-WAN Edge - IDS/IPS Signature Update Succeeded
3. VMware SD-WAN Edge - IDS/IPS Signature Update Failed
image

While we can see them in the Solution details 
image

File: Solutions/VMware SD-WAN and SASE/Analytic Rules/vmw-sdwan-device-congestion.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    lookbackDuration: 5h
+    matchingMethod: AllEntities
+    groupByEntities: []
+    groupByAlertDetails: []
+    groupByCustomDetails: []
+eventGroupingSettings:
+  aggregationKind: SingleAlert
+customDetails:
+  Edge_SerialNr: edgeSerialNumber
+sentinelEntitiesMappings:
```
Comment (v-prasadboke) [inline Solutions/VMware SD-WAN and SASE/Analytic Rules/vmw-sdwan-device-congestion.yaml:39]: Hello @stlaszlo, Im not getting this sentinelEntitiesMappings
You can entity mappings over here

File: Solutions/VMware SD-WAN and SASE/Analytic Rules/vmw-sdwan-idps-update-success.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    groupByAlertDetails: []
+    groupByCustomDetails: []
+eventGroupingSettings:
+  aggregationKind: AlertPerResult
+alertDetailsOverride:
+  alertDescriptionFormat: '{{message}} '
+  alertDynamicProperties: []
+customDetails:
+  idpsSignatureVersion: idpsSignatureVersion
+sentinelEntitiesMappings:
```
Comment (v-prasadboke) [inline Solutions/VMware SD-WAN and SASE/Analytic Rules/vmw-sdwan-idps-update-success.yaml:43]: Same is the case here

File: Solutions/VMware SD-WAN and SASE/Analytic Rules/vmw-sdwan-idps-updatefailed.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    groupByAlertDetails: []
+    groupByCustomDetails: []
+eventGroupingSettings:
+  aggregationKind: AlertPerResult
+alertDetailsOverride:
+  alertDescriptionFormat: '{{message}} '
+  alertDynamicProperties: []
+customDetails:
+  idpsSignatureVersion: idpsSignatureVersion
+sentinelEntitiesMappings:
```
Comment (v-prasadboke) [inline Solutions/VMware SD-WAN and SASE/Analytic Rules/vmw-sdwan-idps-updatefailed.yaml:46]: Over here too

**Comment by stlaszlo** [comment]
@v-prasadboke I will have a look tomorrow as I am on MWC. Thanks for the updates.

**Comment by v-prasadboke** [comment]
Ok Thanks @stlaszlo.

**Comment by v-prasadboke** [comment]
Hello @stlaszlo, Any updates on the above

**Comment by v-prasadboke** [comment]
You can share it over here

**Comment by stlaszlo** [comment]
Hi @v-prasadboke I have attached a ZIP archive of screenshots, I am not sure what are you after, but I've tried attaching screenshots from the App Insights showing logs of successful event retrieval.

One thing to note is that I've done a clean deployment, application files are populated in VS Code, but the functions blade is empty. I've attached a working screenshot (function app name starts with laszlo-) and the new deployment. Could you please check the ZIP archive contents and let me know if I've named the files correctly?
[functionapp_screenshots.zip](https://github.com/Azure/Azure-Sentinel/files/14537831/functionapp_screenshots.zip)

**Comment by v-prasadboke** [comment]
Hello @stlaszlo, unable to pull latest commits to my local. 
Will get back to you by 13 March, 2024. 
Can you confirm once if I'm in the contributor list.

**Comment by stlaszlo** [comment]
Hey @v-prasadboke,

Sure, thanks. Yes, you are still a contributor:
image

**Review by v-prasadboke — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**